### PR TITLE
Adding handling for null block in getBlocksForRange

### PR DESCRIPTION
**Pull request by jpuri**
https://github.com/MetaMask/metamask-extension/issues/10550

PR fixes issue that `getBlocksForRange` can return null block. The approach takes for the fix is as details [here](https://github.com/MetaMask/metamask-extension/issues/10550#issuecomment-1111371195):

1. is a block is not found retry 3 times
3. if block is not found even after 3rd retry propagate error

### CODE REVIEW

File: getBlocksForRange.js [hunk 1]
```diff
@@ -33,11 +33,21 @@ function intToHex(int) {
   return '0x' + hexString
 }
```
Comment (jpuri) [inline getBlocksForRange.js:35]: Not sure if there is a better way to add retry to method `provider.sendAsync`.

Comment (jpuri) [reply getBlocksForRange.js:35]: thanks @mcmire, I made this change

**Review by kumavis — DISMISSED** [review]

File: getBlocksForRange.js [hunk 2]
```diff
@@ -33,11 +33,36 @@ function intToHex(int) {
...
+        method,
+        params,
+      });
+    } catch (error) {
+      console.error(
+        `provider.sendAsync failed: ${error.stack || error.message || error}`
+      );
+    }
+  }
+  throw new Error(`Block not found for params: ${params}`);
```
Comment (mcmire) [inline getBlocksForRange.js:67]: [nit] The params in the error message seem like they get oversimplified. What about `JSON.stringify(params)`?

**Review by mcmire — APPROVED** [review]
One little thing, but looks good regardless.

File: getBlocksForRange.js (hunk 2 above)
Comment (jpuri) [reply getBlocksForRange.js:67]: Good point @mcmire

**Comment by kumavis** [comment]
i think the actual fix is here https://github.com/MetaMask/eth-json-rpc-filters/pull/83

**Pull request merged**